xio-core/src/main/java/com/xjeffrose/xio/config/ConfigReloader.java

```diff
@@ -106,6 +157,15 @@ void setUpdater(BiConsumer<T, T> updater) {
     this.updater = updater;
   }
 
+  public void addWatchFile(File file) throws IllegalStateException {
+    if (!watchFiles.containsKey(file.getAbsolutePath())) {
```

Comment:
this if check (whole function rather) isn't really necessary ... this isn't a MultiMap so only one value per key